recipes-kernel/linux/linux-imx_%.bbappend

```diff
@@ -1,44 +1,11 @@
 FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"
 
-# override kernel source
-SRCBRANCH = "${@bb.utils.contains('SUBMACHINE', 'oppcharge', \
-                    'v4.11.12-evacharge-se-dfs', \
-                    bb.utils.contains('MACHINE', 'tarragon', \
-                        'v6.1-tarragon', \
-                        'v4.9.11', \
-                    d), \
-                d)}"
-
-SRCREV    = "${@bb.utils.contains('SUBMACHINE', 'oppcharge', \
-                    'e428eefb422653dcd5983ff685b9b0972a961b3c', \
-                    bb.utils.contains('MACHINE', 'tarragon', \
-                        'a3c2cea79b7019fe674c4fccc62f54998b509693', \
-                        '87f8fccf0251394fc6fbd5bdddff573230e4e944', \
-                    d), \
-                d)}"
-
-LIC_FILES_CHKSUM = "file://COPYING;md5=${@bb.utils.contains('MACHINE', 'tarragon', \
-                    '6bc538ed5bd9a7fc9398086aedcd7e46', 'd7810fab7487fb0aad327b76f1be7cd7', d)}"
-
-# OppCharge requires a different defconfig from its kernel tree, see KERNEL_DEFCONFIG below
-SRC_URI = "\
-           git://github.com/chargebyte/linux.git;protocol=https;branch=${SRCBRANCH} \
-           ${@bb.utils.contains('SUBMACHINE', 'oppcharge', '', 'file://defconfig', d)} \
-          "
+SRC_URI = "git://github.com/chargebyte/linux.git;protocol=https;branch=${SRCBRANCH}"
 
 LINUX_VERSION_EXTENSION = "-chargebyte"
 
-LINUX_VERSION = "${@bb.utils.contains('SUBMACHINE', 'oppcharge', \
-                        '4.11.12', \
-                        bb.utils.contains('MACHINE', 'tarragon', \
-                            '6.1.49', \
-                            '4.9.11', \
-                        d), \
-                    d)}"
+LIC_FILES_CHKSUM = "file://COPYING;md5=6bc538ed5bd9a7fc9398086aedcd7e46"
 
 KCONFIG_MODE="--alldefconfig"
 
-COMPATIBLE_MACHINE = "(mxs|mx6|mx7|mx8)"
-
-# OppCharge kernel tree contains the required defconfig
-KBUILD_DEFCONFIG = "${@bb.utils.contains('SUBMACHINE', 'oppcharge', 'evacharge-se_defconfig', '', d)}"
+include ${MACHINE}.inc
```

Comment:
Would it be better to move the include all the way up, or does it throw errors?
I checked the Yocto documentation, and I think using `require` instead of `include` is safer, since it throws an error when the file is not found. What do you think?

Reply:
Good point, will change.